src/BinSkim.Rules/PERules/BA2006.BuildWithSecureTools.cs

```diff
@@ -299,93 +241,124 @@
                         string.Join(", ", sorted)));
         }
 
-        private string BuildCompilerIdentifier(ObjectModuleDetails omDetails)
-        {
-            return omDetails.CompilerName + ":" + omDetails.Language + ":" + omDetails.CompilerBackEndVersion;
-        }
-
-        private string BuildMinimumCompilersList(BinaryAnalyzerContext context, Dictionary<Language, List<ObjectModuleDetails>> languageToBadModules)
+        internal void GenerateMessageParametersAndLog(BinaryAnalyzerContext context, Dictionary<Language, List<ObjectModuleDetails>> languageToBadModules)
         {
+            var sb = new StringBuilder();
             var languages = new List<string>();
-
-            foreach (Language language in languageToBadModules.Keys)
+            foreach (KeyValuePair<Language, List<ObjectModuleDetails>> kp in languageToBadModules.OrderBy(l => l.Key))
```

Comment:
>foreach (KeyValuePair<Language, List<ObjectModuleDetails>> kp in languageToBadModules.OrderBy(l => l.Key)) [](http://example.com/codeflow?start=12&length=106)

please clarify this code. in general, I'd prefer not to see computation embedded in a for or foreach construct. 

If we have a need to sort this data, I'd tend to use a data structure that itself sorts keys, please go see if SortedDictionary will serve our purposes here.

Reply:
thanks.
changed to sorted dictionary.